### PR TITLE
Fix Windows MSVC support.

### PR DESCRIPTION
**Pull request by retep998**
Yes, I tested this. Yes, it works.
Also I simplified the API because there's really no reason for the user to customize the naming or directories.
Also made Appveyor test both MSVC and GNU on Windows.

### CODE REVIEW

**Comment by nabijaczleweli** [comment]
Well, I just tested it too and it doesn't, so...

**Comment by nabijaczleweli** [comment]
After changing to "static" linkage again, tho, it works so it might just require "static" on non-MSVC and "dylib" on MSVC.

**Comment by retep998** [comment]
For some reason I'm unable to get my resource file working at all with Mingw. With either `static` or `dylib`.
```
id ICON "Bunny.ico"
```

**Comment by nabijaczleweli** [comment]
Try:

```
#define RT_MANIFEST 24
1 ICON "Bunny.ico" 
```

**Comment by retep998** [comment]
For some reason it still doesn't work. I narrowed it down to the call to `ar`. If I just rename the `foo.res` from windres to `libfoo.a` and just pass that as a library, instead of shelling out to `ar`, it somehow works.

**Comment by nabijaczleweli** [comment]
That does seem to work, yes

File: src/lib.rs [hunk 1]
```diff
@@ -11,9 +11,9 @@
 //! (https://github.com/nabijaczleweli/cargo-update/commit/f57e9c3#diff-a7b0a2dee0126cddf994326e705a91ea),
 //! and now either `windres`-`ar` combo or `RC.EXE` would be used, which was OK.
 //!
-//! Later it transpired, that [MSVC is even more incompatible with everything else]
+//! Later it transpired, that [Rust supports MSVC when not in a VC++ dev environment]
 //! (https://github.com/nabijaczleweli/cargo-update/commit/39fa758#diff-a7b0a2dee0126cddf994326e705a91ea)
-//! by way of not having `RC.EXE` in `$PATH` (because it would only be reasonable to do so),
+//! which means that `RC.EXE` is not in `$PATH` (because that would require a VC++ dev environment),
```
Comment (nabijaczleweli) [inline src/lib.rs]: Don't change Background, please. I's written in a tone that it's supposed to keep.

File: src/windows_not_msvc.rs [hunk 2]
```diff
@@ -7,13 +7,7 @@ pub const SUPPORTED: bool = true;
 pub fn compile_resource(out_dir: &str, prefix: &str, resource: &str) {
     Command::new("windres")
         .args(&["--input", resource, "--output-format=coff", "--output"])
-        .arg(&format!("{}/{}.res", out_dir, prefix))
-        .status()
-        .unwrap();
-
-    Command::new("ar")
-        .args(&["crs", &format!("lib{}.a", prefix), &format!("{}.res", prefix)])
-        .current_dir(&Path::new(&out_dir))
```
Comment (nabijaczleweli) [inline src/windows_not_msvc.rs]: Since you removed `Path` usage, remove `Path` import, too.

File: src/windows_msvc.rs [hunk 3]
```diff
@@ -1,12 +1,13 @@
 use std::path::{PathBuf, Path};
 use std::process::Command;
+use winreg;
```
Comment (nabijaczleweli) [inline src/windows_msvc.rs]: Move to bottom of import list.

**Comment by retep998** [comment]
Did requested changes.

**Comment by nabijaczleweli** [comment]
Right, thanks!

**Comment by nabijaczleweli** [comment]
Released in [`v1.1.0`](https://github.com/nabijaczleweli/rust-embed-resource/releases/tag/v1.1.0)

**Pull request merged**